devito/ir/iet/scheduler.py

```diff
@@ -80,13 +81,15 @@ def _lower_stepping_dims(iet):
         # two different calls to `xreplace`
         mindices = [d for d in i.uindices if d.is_Modulo]
         groups = as_mapper(mindices, lambda d: d.modulo)
+        root = i
         for k, v in groups.items():
             mapper = {d.origin: d for d in v}
-            rule = lambda i: i.function.is_TimeFunction and i.function._time_size == k
-            replacer = lambda i: xreplace_indices(i, mapper, rule)
-            iet = XSubs(replacer=replacer).visit(iet)
+            rule = lambda e: e.function.is_TimeFunction and e.function._time_size == k
+            replacer = lambda e: xreplace_indices(e, mapper, rule)
+            i = XSubs(replacer=replacer).visit(i)
```

Comment:
I just remembered something. Overriding iteration variables in Python can lead to all sort of breakages. 

How about instead:

```
for ...:
    ...
    root = XSubs(replacer=replacer).visit(root)
subs[i] = root
```

?

Reply:
Yeah, that really makes sense!!